ppmclibs/tinycv_impl.cc

```diff
@@ -333,22 +330,22 @@ Image* image_read(const char* filename)
 
 Image* image_from_ppm(const unsigned char* data, size_t len)
 {
-    vector<uchar> buf(data, data + len);
+    std::vector<uchar> buf(data, data + len);
     Image* image = new Image;
-    image->img = imdecode(buf, CV_LOAD_IMAGE_COLOR);
+    image->img = cv::imdecode(buf, CV_LOAD_IMAGE_COLOR);
     return image;
 }
 
-bool image_write(Image* s, const char* filename)
+bool image_write(Image* const s, const char* filename)
```

Comment:
Making arguments `const` is actually not common. Or did you intend making the pointer target const (`const Image*  s`)?

Reply:
I think preventing arguments from being changed is a good practice also good point I think making the pointer const would be usefull as well..

Reply:
I think in this case removing using std and cv is good idea especially because you use std::vector and opencv vectors. Since this is all in one file it is making code better readable, understandable preventing programing errors in the future as well as namespace clashes.

Reply:
I agree with @Amynka - namespace clashes aren't fun

Reply:
> because you use std::vector and opencv vectors

Ok, that's a good reason.

---

> making the pointer const would be usefull as well

Making the pointer *itself* const is actually not bad, but it is common exception to make function arguments not const (still talking about the pointer *itself*): https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con1-by-default-make-objects-immutable  
If you want to make it const nevertheless, at least be consistent and make the pointer to filename also const.